### PR TITLE
Add support for App Clips

### PR DESCRIPTION
**Pull request by brentleyjones**
Embeds App Clips into the containing app when they are a dependency.

### CODE REVIEW

**Comment by brentleyjones** [comment]
@yonaskolb This is an Xcode 12 only feature. How do we want to go about testing this, since the figure will fail when tested with Xcode 11.

**Comment by dflems** [comment]
@brentleyjones: Maybe `project.yml` could be templated (with mustache or some other template engine) to selectively ignore certain sections of the file unless a particular version of Xcode is selected?

**Comment by dflems** [comment]
I fiddled with it a little locally. This might require a new generated project file :\

**Comment by brentleyjones** [comment]
@yonaskolb FYI, we aren't actually testing this with CI because of the Spectre issue, but all of the infrastructure is in place for when we do add it.
```yaml
xcode: ["11"] # TODO: add 12_beta (blocked by https://github.com/kylef/Spectre/issues/41)
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #### Added
 - Add `onlyCopyFilesOnInstall` option to targets for the Embed Files build phase. [#912](https://github.com/yonaskolb/XcodeGen/pull/912) @jsorge
+- Adds App Clip support. [#909](https://github.com/yonaskolb/XcodeGen/pull/909) @brentleyjones
```
Comment (dflems) [inline CHANGELOG.md:7]: @brentleyjones Can you tag me in on this as well? 😄

Comment (brentleyjones) [reply CHANGELOG.md:7]: Yep!

**Comment by brentleyjones** [comment]
@yonaskolb This is green now.

**Comment by dflems** [comment]
🙌 

Thanks @yonaskolb!

w.r.t the Spectre issue: in Xcode 12, `XCTest` now tracks with the latest version of macOS SDK but the SwiftPM default deployment target is `10.10` so test libraries that use `XCTest` get a little... weird. Gonna take another stab at trying to properly fix today.

**Comment by dflems** [comment]
@yonaskolb @brentleyjones 

It looks like both Spectre and XcodeGen build fine on Xcode 12 beta 4 but GitHub Actions doesn't support it yet. They're currently targeting build `12A8169g`, which is beta 3 ([link](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md)). Not sure when they're planning on supporting it, but once that's upgraded, we should be able to turn on the CI that will exercise these new codepaths.

**Pull request merged**